android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -1662,4 +1663,29 @@ class QuestionnaireViewModelTest : RobolectricTest() {
     Assert.assertEquals(formatter.format(group.meta.lastUpdated), formatter.format(Date()))
     unmockkObject(ResourceMapper)
   }
+
+  @Test
+  fun `updateResourceLastUpdatedLinkedAsSubject() should catch ResourceNotFoundException`() {
+    val resourceNotFoundException =
+      ResourceNotFoundException(type = "Resource not found exception", id = "1234")
+    val questionnaireResponse =
+      QuestionnaireResponse().apply {
+        id = "1234"
+        subject.reference = "Patient/123467"
+      }
+    coEvery { defaultRepo.loadResource(any(), any()) } throws resourceNotFoundException
+    coEvery {
+      questionnaireViewModel.updateResourceLastUpdatedLinkedAsSubject(questionnaireResponse)
```

Comment:
if we're mocking the whole function we're not going to test any of the changed lines, need to mock further in